### ISSUE TITLE
Simplify single node installs

### ISSUE THREAD

**Issue by WesleyW** [yugabyte/yugabyte-db#1142]
## Goal
Simplify single node YugaByte DB installs by unifying yb-master and yb-tserver under one parent process. This will allow managing YugaByte DB (actions such as starting and stopping the database daemon) to interact with only one process. This change will also unify YB processes under one process - killing `yugabyte-db` will also kill any master/tserver processes. 

## Description
Currently, `yb-ctl` starts independent master and tserver processes in the background. It would be better to have one managing process (`Yugabyte-DB`). Primarily, this would allow us to run Yugaware, Yugabyte, and any other processes with one command. Additionally, we will move towards the usage of `master.conf` and `tserver.conf` configuration files instead of passing in flags to `yugabyte-db`. Directory locations are TBD.

### Starting
`yugabyte-db start` will start `yb-master` and `yb-tserver` in the background. `yugabyte-db` will continue running as a daemon which will restart either master or tserver processes if they die.

### Stopping
`yugabyte-db stop` will stop `yb-master`, `yb-tserver`, and `yugabyte-db`. Data will be preserved.

### Restarting
`yugabyte-db restart` will stop all YB processes (including `yugabyte-db`) and restart them.

~~### Destroying~~
~~`yugabyte-db destroy` will stop all YB processes as well as destroy any data which was created.~~

**Comment by rkarthik007** [comment]
Do we really need the `yugabyte-db destroy` command for a single node install? This seems a bit off, since there would be nothing to "destroy" as such. Feels like the "stop" command will stop all processes. Maybe this should be an "uninstall" command? cc @bmatican @schoudhury

**Comment by WesleyW** [comment]
“Uninstall” to me means they don’t have Yugabyte itself anymore. “Destroy” or something similar deletes data but doesn’t uninstall.

**Comment by rkarthik007** [comment]
I don't think this option is easy to understand. I would rather the sequence be as follows:
- Stop the db to make sure all processes are down
- Give instructions to manually delete the data folders (rm -rf the data)

I agree this may come in handy, but lets introduce it as an "advanced" flag by having the end user pass a "--force" option or something equivalent (and lets document it as an advanced flag also). For now, I would propose dropping the destroy functionality all together from here. Does that work?

**Comment by kmuthukk** [comment]
@WesleyW - btw, in this model will there be a notion of a conf directory as well (which is separate from the data directory)-- and will we move to a yb-master.conf and yb-tserver.conf file for any non-defaulted flags we want to pass (including things like whether bind addresses are loopback addresses or not, etc.)?

**Comment by WesleyW** [comment]
Seems reasonable @rkarthik007 . Good point, @kmuthukk. We're moving to using only configuration files now, ammended issue to reflect that.

**Comment by kriteshranjan** [comment]
Closing this issue based on AI validation. cc: @m-iancu , @VamsidharGunturu
